packages/playground/blueprints/src/lib/steps/login.ts

```diff
@@ -37,22 +38,34 @@ export const login: StepHandler<LoginStep> = async (
 	progress
 ) => {
 	progress?.tracker.setCaption(progress?.initialCaption || 'Logging in');
-	// Allow WordPress to set the cookies.
-	await playground.request({
-		url: '/wp-login.php',
-	});
 
+	const docroot = await playground.documentRoot;
+
+	// Login as the current user without a password
+	await playground.writeFile(
+		`${docroot}/playground-login.php`,
+		`<?php
+		require_once( dirname( __FILE__ ) . '/wp-load.php' );
+		if ( is_user_logged_in() ) {
+			return;
+		}
+
+		$user = wp_signon( array(
+			'user_login'    => ${phpVar(username)},
+			'user_password' => ${phpVar(password)},
+			'remember'      => true
+		) );
```

Comment:
If we do this on the PHP side we could remove the need to use a password and use `wp_set_auth_cookie()` instead of `wp_signon()`. It's not a big deal but with this we are storing the password in clear text.

Reply:
I agree, there is a working script in the PR description that would remove the need for passwords.

I'm happy to do it, but I wanted to get feedback first.

@adamziel what do you think about switching to passwordless logins?

Reply:
Oh sorry, I had missed that!

Reply:
This came up in another issue, see:

https://github.com/WordPress/wordpress-playground/pull/1753#discussion_r1755602424

Tl;dr let's avoid writing to document root and make it a flexible mu-plugin for all runtimes. That would effectively be a passwordless login

Reply:
> Tl;dr let's avoid writing to document root and make it a flexible mu-plugin for all runtimes. That would effectively be a passwordless login

I'm having problems with [making this approach work](https://github.com/WordPress/wordpress-playground/pull/1753#discussion_r1755602424). I assume it's because it's triggered by page loads.

I ended up using my implementation with a HTTP request but it uses auto_prepend_file to add the file instead of writing it inside the step.

Reply:
> I'm having problems with making this approach work. I assume it's because it's triggered by page loads.

What are these problem?